docs/spec/spec.md

```diff
@@ -323,6 +323,48 @@ Here the returned stream can consist of following types of records,
    sql:ParameterizedQuery query = `SELECT * FROM students WHERE id < ${id} AND age > ${age}`;
    stream<Student, sql:Error?> resultStream = dbClient->query(query);
    ```
+   
+   `sql:Column` annotation can be used to map database columns to record fields of different name. This annotation should be attached to record fields.
+   ```ballerina
+   type Student record {
+       int id;
+       @sql:Column { name: "first_name" }
+       string firstName;
+       @sql:Column { name: "last_name" }
+       string lastName
+   };
+   ```
+   The above annotation will map the database column `first_name` to the Ballerina record field `firstName`. If the `query()` function does not return `first_name` column, the field will not be populated.
+
+   Multiple table columns can be matched to a single Ballerina record within a returned record. For instance if the query returns data from multiple tables as follows,
+   ![schema](../proposals/resources/schema1.png)
+   Both `TEACHERS.id` and `TEACHERS.name` can be grouped to another Typed record such as `Teacher` type.
+
+   ```ballerina
+   public type Students record {|
+       int id;
+       string name;
+       string? age;
+       float? gpa;
+       Teachers teachers;
+   |}
+   type Teachers record {|
+       int id;
+       string name;
+   |}
+   ```
+   In the above scenario also, `sql:Column` annotation can be used to rename field name such as,
+   ```ballerina
+   public type Students record {|
+       int id;
+       string name;
+       string? age;
+       float? gpa;
+       @sql:Column {
+           name: "teachers"
+       Teacher teacher;
```

Comment:
```suggestion
       @sql:Column { name: "teachers" }
       Teacher teacher;
```

Reply:
Fixed in [482129a](https://github.com/ballerina-platform/module-ballerina-sql/pull/509/commits/482129a67c0b9df3351f4db99b523cea328695dd)